src/messages/manifestjson.js

```diff
@@ -51,6 +51,23 @@ export function manifestFieldUnsupported(fieldName, error) {
   };
 }
 
+export const MANIFEST_FIELD_PRIVILEGED = 'MANIFEST_FIELD_PRIVILEGED';
+export function manifestFieldPrivileged(error) {
+  const messageTmpl =
+    i18n._(oneLine`%(instancePath)s: privileged manifest fields
+                     are only allowed in add-ons signed as privileged.`);
+  const message = i18n.sprintf(messageTmpl, {
+    instancePath: error.instancePath,
+  });
+
+  return {
+    code: MANIFEST_FIELD_PRIVILEGED,
+    message,
+    description: message,
```

Comment:
I think the `description` should include a bit more information so I'd reuse `message` + the following:

```
Please refer to https://github.com/mozilla-extensions/xpi-manifest to learn more about privileged extensions and signing.
```